pkg/apis/resource/validation/validation.go

```diff
@@ -87,6 +92,18 @@ func validateClassParameters(ref *resource.ResourceClassParametersReference, fld
 	return allErrs
 }
 
+func validateAPIGroup(group string, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if group != "" {
+		if errs := utilvalidation.IsDNS1123Subdomain(group); len(errs) > 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath, group, strings.Join(errs, ",")))
+		} else if len(strings.Split(group, ".")) < 2 {
```

Comment:
Is `apps` a vaild group name in XXXParametersRef? I copied the validation logic from crd. https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go#L349

cc @pohly

Reply:
A better in-tree reference might be this:
https://github.com/kubernetes/kubernetes/blob/c8a51aaffc94da6e7a66d8984890d3111e361a54/pkg/apis/networking/validation/validation.go#L550-L570

"apps" is not valid, it's "apps.k8s.io". But the empty string is allowed, it stands for the core API. It's just not allowed for a CRD.

Reply:
@pohly updated.

BTW, `apps` is a valid DNS1123 subdomain.

Reply:
True. So the validation isn't as strict as it needs to be for this particular case, but that's okay. It doesn't have to catch everything.